testing/performance_testing/uto/TestStrategy.md

```diff
@@ -0,0 +1,190 @@
+- **Description**: This test case simulates a scenario where Alice needs to create hundreds of topics, perform parallel data ingestion, consume data from these topics simultaneously, and then delete all the topics. The primary focus is on minimizing latency during bulk create and delete operations and ensuring system stability under heavy load.
+- **Preconditions**:
+  * A Kafka cluster with the UTO deployed is available.
+  * Necessary tools and scripts for creating topics, producing and consuming messages are set up.
+  * Monitoring tools for tracking performance metrics are in place.
+- **Test Steps**:
+  1. Topic Creation:
+      * Create a script to generate and execute requests for creating hundreds of topics in Kafka.
+      * Ensure that these topic creation requests are made in a bulk operation.
+  2. Data Ingestion:
+      * Simulate data production to all created topics simultaneously. Use a script to send a large volume of messages to these topics.
+  3. Data Consumption:
+      * Simultaneously with the ingestion, start consuming messages from all the topics.
+      * Ensure the consumption process is parallelized to mimic real-world batch processing.
+  4. Topic Deletion:
+      * After a specified period or once a certain amount of data is ingested and consumed, initiate the deletion of all topics.
+      * Monitor the time taken and system behavior during this bulk deletion process.
+  5. Monitoring and Data Collection:
+      * Throughout the test, monitor and record relevant metrics such as throughput, latency, resource utilization (CPU, memory, I/O), and error rates.
+      * Pay special attention to the latency during the bulk creation and deletion of topics.
+        Expected Results:
+     * Topics are created, and data is ingested and consumed as expected without significant delays.
+     * The system maintains stability and performance throughout the test.
+     * Latency for bulk operations (creation and deletion) is within acceptable limits.
+     * Resource utilization stays within expected thresholds, indicating no bottlenecks.
+       Post-Conditions:
+     * All test topics are successfully deleted.
+     * System returns to normal operational status.
+- **Variations**:
+  * Vary the number of topics and the volume of data to test system limits.
+  * Adjust batch size and linger time to evaluate their impact on performance.
+- **Risks and Mitigation Strategies**:
+  * System overload: Monitor resource utilization closely and have fallbacks or thresholds to prevent system crashes.
+
+**Note:** that this table is just filled with random values. We could definitely extract or add some metrics, which we think would be great for our findings.
+
+| Experiment # | Batch Size | Linger Time (s) | Topics Created | Throughput (msgs/sec) | Latency (ms) | CPU Utilization (%) | Memory Utilization (%) | Experiment Notes           |
```

Comment:
I think we are interested in throughput and latency of topic events, but here an in the following test case 2 looks like we are tracking messages instead. Do we want to track per topic event type?